irohad/simulator/impl/simulator.cpp

```diff
@@ -65,8 +66,10 @@ namespace iroha {
         const shared_model::interface::Proposal &proposal) {
       log_->info("process proposal");
       // Get last block from local ledger
-      block_queries_->getTopBlocks(1).as_blocking().subscribe(
-          [this](auto block) { last_block = block; });
+      block_queries_->getTopBlocks(1)
+          .subscribe_on(rxcpp::observe_on_new_thread())
```

Comment:
May you explain why should it be on a separate thread?

Reply:
It is a bug, related to calling simulator from the same thread which waits on simulator, which leads to hanging since we are blocked. You can ask Andrei for details

Reply:
Otherwise, we need to synchronize access from separate threads.